renderdoc/driver/gl/gl_debug.cpp

```diff
@@ -1044,18 +1044,38 @@ uint32_t GLReplay::PickVertex(uint32_t eventID, const MeshDisplay &cfg, uint32_t
 
     PickMVP = projMat.Mul(camMat.Mul(guessProj.Inverse()));
   }
+  
+  vec3 rayPos;
+  vec3 rayDir;
+  // convert mouse pos to world space ray
+  {
+	  Matrix4f InversePickMVP = PickMVP.Inverse();
+
+	  float pickX = ((float)x) / ((float)DebugData.outWidth);
+	  float pickXCanonical = (1.0f - pickX) * -1 + pickX * 1;// go from 0 1 range to "-1 1" range, is there a lerp function somewhere?
```

Comment:
There's no lerp function but feel free to add it now that you need it. I think a `Lerp` function in `common.h` along with others like `AlignUp` is probably the right place to put it, until there are enough math functions to justify a separate header.